### PR TITLE
Add an example of option pricing using Monte-Carlo simulation

### PR DESCRIPTION
**Pull request by rezoo**
Merge #473 first.
This PR computes the price of a call option using Monte-Carlo simulation. In my environment (GPU: TITAN X(Pascal)), the performance is the following:

```
python monte_carlo.py 
initializing...
start computation
    # of options: 1000
    # of samples per option: 100000000
GPU (CuPy, Monte Carlo method): 39.067884 sec
Error: 0.007679
```

### CODE REVIEW

File: examples/finance/monte_carlo.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    // In this case, we used an index as a seed of random sequence.
+    uint64_t rand_state[2];
+    init_state(rand_state, i, seed);
+
+    T call_sum = 0;
+    const T v_by_sqrt_t = v * sqrt(t);
+    const T mu_by_t = (r - v * v / 2) * t;
+
+    // compute the price of the call option with Monte Carlo method
+    for(int i=0; i<n_samples; ++i) {
```
Comment (unnonouno) [inline examples/finance/monte_carlo.py]: Insert spaces:
```
for (int i = 0; i < n_samples; ++i) {
```

File: examples/finance/monte_carlo.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        const uint64_t x = xorshift128plus(state);
+        // 18446744073709551615 = 2^64 - 1
+        return T(x) / T(18446744073709551615);
+    }
+
+    // Draw a sample from a normal distribution with N(0, 1)
+    __device__ inline T sample_normal(uint64_t* state) {
+        T x = sample_uniform(state);
+        T s = T(-1.4142135623730950488016887242097);  // = -sqrt(2)
+        if(x > 0.5) {
```
Comment (unnonouno) [inline examples/finance/monte_carlo.py]:
```
if (x > 0.5) {
```

File: examples/finance/monte_carlo.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    # Because of the broadcasting rule, in this case this kernel
+    # launches n_options * n_threads_per_options threads
+    # each of which corresponds to the element of 'call_prices'.
+    monte_carlo_kernel(
+        stock_price[:, None], option_strike[:, None], option_years[:, None],
+        risk_free, volatility, n_samples_per_thread, seed, call_prices)
+    return call_prices.mean(axis=1)
+
+
+if __name__ == '__main__':
```
Comment (unnonouno) [inline examples/finance/monte_carlo.py]: Can you make a `main` function and call it here not to use global scope?

**Comment by rezoo** [comment]
Thanks. I've fixed.

**Comment by rezoo** [comment]
@unnonouno Hi, is there any progress?

**Comment by unnonouno** [comment]
It looks good. Can you add a test to run it with small data? https://github.com/cupy/cupy/tree/master/tests/example_tests

**Comment by rezoo** [comment]
I added tests.

File: tests/example_tests/test_finance.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import unittest
+
+import six
+
+import example_test
```
Comment (unnonouno) [inline tests/example_tests/test_finance.py]: `from example_tests import example_test`

Comment (rezoo) [reply tests/example_tests/test_finance.py]: Fixed.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**